From: NeilBrown <neilb@suse.com>
Subject: [PATCH] lockd: use init_utsname for id encoding
Patch-mainline: never, they have more complex patch for more complex need.
References: bsc#1033804

Taking hostname from uts namespace if not safe, because this cuold be
performind during umount operation on child reaper death. And in this case
current->nsproxy is NULL already.

So use init_utsname() instead.  That is always safe.
In 3.0, sunrpc and nfs are not name-space aware, so depending on lockd
knowing the host name of a particular container is not going to work.
So there is no cost in just using init_utsname().

Signed-off-by: NeilBrown <neilb@suse.com>

---
 fs/lockd/mon.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/fs/lockd/mon.c
+++ b/fs/lockd/mon.c
@@ -429,7 +429,7 @@ static void encode_my_id(struct xdr_stre
 {
 	__be32 *p;
 
-	encode_nsm_string(xdr, utsname()->nodename);
+	encode_nsm_string(xdr, init_utsname()->nodename);
 	p = xdr_reserve_space(xdr, 4 + 4 + 4);
 	*p++ = cpu_to_be32(argp->prog);
 	*p++ = cpu_to_be32(argp->vers);
